firestore.rules

```diff
@@ -389,19 +413,19 @@ service cloud.firestore {
         }
         match /comments/{commentId} {
           // portal-authenticated teachers with access to the document can create valid comments
-          allow create: if teacherCanAccessDocument() && isValidCommentCreateRequest();
+          allow create: if (teacherCanAccessDocument() || researcherCanAccessDocument()) && isValidCommentCreateRequest();
           // teachers can only update their own comments and only if they're valid
-          allow update: if teacherCanAccessDocument() && isValidCommentUpdateRequest();
+          allow update: if (teacherCanAccessDocument() || researcherCanAccessDocument()) && isValidCommentUpdateRequest();
           // teachers can only delete their own comments
-          allow delete: if teacherCanAccessDocument() && userIsResourceUser();
+          allow delete: if (teacherCanAccessDocument() || researcherCanAccessDocument()) && userIsResourceUser();
           // only teachers that "own" the document can read the comments (for now)
-          allow read: if teacherCanAccessDocument();
+          allow read: if teacherCanAccessDocument() || researcherCanAccessDocument();
```

Comment:
I think because `researcherCanAccessDocument()` looks at `getDocumentData().context_id` it means whenever a researcher does a query for a list of comments they'll have to filter by the `context_id`. This makes sense to do, I just want to point it out because it might require updating some queries in CLUE.